### ISSUE TITLE
How user get newly added Deb info ?

### ISSUE THREAD

**Issue by kcubeterm** [termux/whatprovides#5]
Do we need to implement "update or -u" option to update the files in DB_PATH or is it okay to release repo on every two days.

**Comment by ghost** [comment]
Right now we can do just periodic updates of the package.

Final solution would be use of Github Actions to generating the DB and uploading it to Bintray on periodic basis. Preferably to do that once per week to reduce bandwidth usage as we will need to download full set of packages for `all` and at least `aarch64`. Also I suggest to switch to normal DB format like SQLite. -- will allow to implement `whatprovides --update`. *Depends on https://github.com/termux/whatprovides/issues/4.*

**Comment by Grimler91** [comment]
> Right now we can do just periodic updates of the package.

> Final solution would be use of Github Actions to generating the DB and uploading it to Bintray on periodic basis. Preferably to do that once per week to reduce bandwidth usage as we will need to download full set of packages for all and at least aarch64. Also I suggest to switch to normal DB format like SQLite. -- will allow to implement whatprovides --update. Depends on #4.

This would also be a great for command-not-found, as we can then automatically re-generate those lists as well.

To save bandwidth we could also use the github CI artifacts, and add a new task before the deb upload step which generates a list from the deb. Getting it to work as part of the CI might be a bit more work than setting up a weekly job that downloads debs and generates lists though, so could be considered a possible improvement for the future.

**Comment by ghost** [comment]
Generating a file list before publishing the debfile would be better as we will not need to download all debfiles.

I have made a SQLite database from bunch of files in `./db` folder: [whatprovides.db.gz](https://github.com/termux/whatprovides/files/5570456/whatprovides.db.gz) (`gunzip` it before use). This will allow to get rid of lots of files and will make update easier. Needs `sqlite` binary as dependency.

Example on how to retrieve data from the db:
```
~ $ echo "SELECT owned_file FROM 'whatprovides' WHERE package_name == 'gitea';" | sqlite whatprovides.db
/.
/data
/data/data
/data/data/com.termux
/data/data/com.termux/files
/data/data/com.termux/files/usr
/data/data/com.termux/files/usr/etc
/data/data/com.termux/files/usr/etc/gitea
/data/data/com.termux/files/usr/etc/gitea/app.ini
/data/data/com.termux/files/usr/bin
/data/data/com.termux/files/usr/bin/gitea
/data/data/com.termux/files/usr/var
/data/data/com.termux/files/usr/var/lib
/data/data/com.termux/files/usr/var/lib/gitea
/data/data/com.termux/files/usr/var/log
/data/data/com.termux/files/usr/var/log/gitea
/data/data/com.termux/files/usr/share
/data/data/com.termux/files/usr/share/doc
/data/data/com.termux/files/usr/share/doc/gitea
/data/data/com.termux/files/usr/share/doc/gitea/LICENSE
```

**Comment by ghost** [comment]
DB is now stored at https://dl.bintray.com/termux/metadata/whatprovides-db/whatprovides.db.gz. Package version 1.0.0 adds support for `whatprovides -u` to fetch the database from Bintray.

CI steps for updating and pushing db to Bintray are not implemented currently.
